RosNet/DataModel/Time.cs

```diff
@@ -29,5 +29,29 @@ public DateTime ToDateTime()
             double secs = this.Secs + this.NSecs * Math.Pow(10, -9);
             return DateTime.UnixEpoch.AddSeconds(secs);
         }
+
+        /// <summary>
+        /// Compares two time objects by comparing sec and nano sec
+        /// </summary>
+        public int CompareTo(object? other)
+        {
+            if (other == null)
+                return 1;
+
+            Time otherTime = other as Time;
+
+            if (otherTime == null)
+                throw new ArgumentException("Other is not of type Time");
+
+            if (this.Secs - otherTime.Secs != 0)
+            {
+                return (int)(this.Secs - otherTime.Secs);
+            }
+            else if (this.NSecs - otherTime.NSecs != 0)
+            {
+                return (int)(this.NSecs - otherTime.NSecs);
+            }
+            return 0;
```

Comment:
```suggestion
            return (int)(this.Secs != otherTime.Secs ? this.Secs - otherTime.Secs : this.NSecs - otherTime.NSecs)
```

Reply:
If `otherTime` is larger than `this`, it would underflow by `uint`-semantics, thus causing a quite nasty bug

Reply:
I did think about that, but seems i forgot before pushing it. Also your way was much cleaner